jacksono/onyango
# Complete backend and front end

### Background context for this PR
Sometimes you just want to take quick note and save it somewhere. This PR adds features that will help you do just that

#### What does this PR do?
* Adds CRUD actions for notes
* Adds user registration, sign in and sign out
* Implements Authorisation and Authentication
* Add front end and back end tests

#### Screenshots

<img width="1440" alt="screen shot 2018-08-29 at 22 45 46" src="https://user-images.githubusercontent.com/4943363/44811486-7f7fbc80-abdd-11e8-9076-4949d63b3e3e.png">
<img width="1440" alt="screen shot 2018-08-29 at 22 45 52" src="https://user-images.githubusercontent.com/4943363/44811488-80185300-abdd-11e8-9138-df33c3f98ccc.png">
<img width="1440" alt="screen shot 2018-08-29 at 22 46 16" src="https://user-images.githubusercontent.com/4943363/44811489-80185300-abdd-11e8-913d-58aeac88933e.png">
<img width="1440" alt="screen shot 2018-08-29 at 22 46 26" src="https://user-images.githubusercontent.com/4943363/44811490-80185300-abdd-11e8-98c0-5c08bc9af13d.png">
<img width="1440" alt="screen shot 2018-08-29 at 22 46 46" src="https://user-images.githubusercontent.com/4943363/44811491-80b0e980-abdd-11e8-804f-19d6d8558e4b.png">
<img width="1440" alt="screen shot 2018-08-29 at 22 46 57" src="https://user-images.githubusercontent.com/4943363/44811494-80b0e980-abdd-11e8-94c1-654e5eba2e22.png">

#### How should PR reviewers test this?
- Run the unit tests with `npm test` and `npm run api-test`
- Clone this branch, navigate to the root folder and run `npm start` then test functionality on the web app
- Use post man to test the end points. See the readme for the available endpoints

#### Would you like specific feedback on anything?
* How is the experience
* Did you find any bugs?
